packages/hashi/src/mainClient.js

```diff
@@ -59,6 +63,18 @@ export default class MainClient {
       });
     });
     this.mediator.sendMessage({ nameSpace, event: events.READYCHECK, data: true });
+
+    this.on(this.events.DATAREQUESTED, message => {
+      this.__fetchContentData(message);
```

Comment:
I am surprised you have not needed to call `.bind` on any of the handlers you have here.